deploy-service/common/src/main/java/com/pinterest/deployservice/handler/DeployHandler.java

```diff
@@ -395,6 +395,13 @@ private void validateBuild(EnvironBean envBean, String buildId) throws Exception
             throw new DeployInternalException("Build name (%s) does not match stage config (%s).",
                     buildBean.getBuild_name(), envBean.getBuild_name());
         }
+        
+        if(envBean.getStage_type() != EnvType.DEV &&
+                buildAllowlist != null && !buildAllowlist.trusted(buildBean.getArtifact_url())
+                &&  !buildBean.getScm_branch().equalsIgnoreCase("private")) {
```

Comment:
```suggestion
                && !buildBean.getScm_branch().equalsIgnoreCase("private")) {
```
nit: remove extra space

Reply:
you want the opposite to prevent NPE.
` "private".equalsIgnoreCase(buildBean.getScm_branch())`

Reply:
Then it's the best to create a constant for `"private"`.

Reply:
I founds that pinboard uses `private_build` instead of `private`. It means they will bypass this check. We need to investigate more.